vllm/entrypoints/openai/serving_chat.py

```diff
@@ -49,23 +57,46 @@ async def create_chat_completion(
             return self.create_error_response(
                 "logit_bias is not currently supported")
 
-        try:
-            prompt = self.tokenizer.apply_chat_template(
-                conversation=request.messages,
-                tokenize=False,
-                add_generation_prompt=request.add_generation_prompt)
-        except Exception as e:
-            logger.error(
-                f"Error in applying chat template from request: {str(e)}")
-            return self.create_error_response(str(e))
+        if "chatglm3" in request.model:
+            [*history, query] = request.messages
+            query = query["content"]
+            prompt_ids = (self.tokenizer.build_chat_input(
+                query=query, history=history,
+                role="user").input_ids.squeeze().tolist())
+            prompt = self.tokenizer.decode(prompt_ids)
```

Comment:
i don't think we need to decode it again except for debugging purpose

Reply:
I use prompt_ids obtained directly from the interface, but subsequent interface calls need to pass in a prompt, so it is not appropriate to generate either one randomly because I am not sure if inconsistency between the two will cause trouble